packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx

```diff
@@ -171,25 +170,49 @@ export const EditTagMenu = ({
       },
       items,
     } satisfies Partial<MenuProps>;
-  }, [legacyProperties, navigate, t, tag]);
+  }, [
+    legacyProperties.workspaceId,
+    navigate,
+    t,
+    tag,
+    tagColor,
+    tagService,
+    tagValue,
+  ]);
 
   return <Menu {...menuProps}>{children}</Menu>;
 };
 
 export const TagsEditor = ({
+  pageId,
   options,
   value,
-  onChange,
-  onOptionsChange,
```

Comment:
I believe you do not need to change TagsEditor & InlineTagsList at all. You can just wrap the TagService into onChange & onOptionsChange where TagsEditor is being used

Reply:
Portability-wise it might not be a good idea to call entity methods deeply inside of an UI view.
IMO it might be worth adding a glue layer that converts Tag entity into the required callbacks/APIs that required by TagsEditor. If there is anything that is not that well fit into good CRDT practices, we can make changes to the exposed props of TagsEditor instead, like exposing various callbacks like `onCreateTag`, `onRenamingTag` etc

Reply:
Oh, I agree with this.. 

But there's also the question here is we should lifting up the side effect to which level.

If it is just lift up to PagePropertyValue (the upper level of this file), it seems to be no big difference.

It would be nice to lift up to a higher level and make the PageProperty pure.

But currently our tag is not stored as a regular page property, so it is not easy to implement..